recipes/flatc/all/conanfile.py

```diff
@@ -22,11 +22,16 @@ def _source_subfolder(self):
     @property
     def _build_subfolder(self):
         return "build_subfolder"
-
+    
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+            
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         extracted_dir = "flatbuffers-" + self.version
         os.rename(extracted_dir, self._source_subfolder)
+        self._patch_sources()
```

Comment:
Please add this call to the `build()` method.
We like to keep the sources clean in the source directory.

Reply:
Clear, fixed. In the conan documentation it is stated that it is mostly used in the sources step. Is that something that should be updated?

https://docs.conan.io/en/latest/reference/tools.html#tools-patch

Reply:
This is mostly a convention in this repo.
pros are:
- apply different patches for different build configurations
- not having to re-download/re-extract sources for every build (if you don't have the download cache enabled)
cons are:
- not being able to use `no_copy_source` and having to copy the source tree each time for every build

@uilianries  Your input?

Reply:
Clear, sounds sensible.

Reply:
@madebr I should update the documentation.

Reply:
PR: https://github.com/conan-io/docs/pull/1605